keylime/tenant.py

```diff
@@ -79,82 +93,61 @@ def __init__(self):
+                if not os.path.isfile(key):
+                    logger.warning("Could not find file %s provided in 'client_key'", key)
+
+                if not trusted_ca:
+                    logger.warning("No certificates provided in 'trusted_server_ca'")
+                else:
+                    self.client_cert = cert
+                    self.client_key = key
+                    self.client_key_password = key_password
+                    self.verify_server_cert = verify_server_cert
+                    self.trusted_server_ca = trusted_ca
+
+                    self.tls_context = web_util.generate_tls_context(
+                        cert, key, trusted_ca, key_password, verify_server_cert, is_client=True, logger=logger
+                    )
+
+                    logger.info("TLS is enabled.")
 
     @property
     def verifier_base_url(self):
         return f"{self.verifier_ip}:{self.verifier_port}"
 
-    @staticmethod
-    def get_tls_context():
-        """Generate certifcate naming and path
-
-        Returns:
-            string -- my_cert (client_cert), my_priv_key (client private key)
-        """
-        verifier_ca_cert = config.get("tenant", "ca_cert")
-        my_cert = config.get("tenant", "my_cert")
-        my_priv_key = config.get("tenant", "private_key")
-        tls_dir = config.get("tenant", "tls_dir")
-
-        if tls_dir == "default":
-            verifier_ca_cert = "cacert.crt"
-            my_cert = "client-cert.crt"
-            my_priv_key = "client-private.pem"
-            tls_dir = "cv_ca"
-
-        if tls_dir[0] != "/":
-            tls_dir = os.path.abspath(os.path.join(config.WORK_DIR, tls_dir))
-
-        logger.info("Setting up client TLS in %s", tls_dir)
-        verifier_ca_cert = os.path.join(tls_dir, verifier_ca_cert)
-        my_cert = os.path.join(tls_dir, my_cert)
-        my_priv_key = os.path.join(tls_dir, my_priv_key)
-
-        tls_context = (my_cert, my_priv_key)
-
-        agent_mtls_context = (None, None)
-        # Check for user defined CA to connect to agent
-        agent_mtls_cert_enabled = config.getboolean("tenant", "agent_mtls_cert_enabled", fallback=False)
-
-        if agent_mtls_cert_enabled:
-            agent_mtls_cert = config.get("cloud_verifier", "agent_mtls_cert", fallback=None)
-            agent_mtls_private_key = config.get("cloud_verifier", "agent_mtls_private_key", fallback=None)
-            agent_mtls_context = tls_context
-            if agent_mtls_cert != "CV":
-                agent_mtls_context = (agent_mtls_cert, agent_mtls_private_key)
-
-        return tls_context, agent_mtls_context, verifier_ca_cert
-
     def process_allowlist(self, args):
         # Set up PCR values
-        tpm_policy = config.get("tenant", "tpm_policy")
         if "tpm_policy" in args and args["tpm_policy"] is not None:
             tpm_policy = args["tpm_policy"]
+        else:
+            # Use default empty TPM policy
+            tpm_policy = "{}"
```

Comment:
All the block can be replaced with `tpm_policy = args.get("tpm_policy") or {}`. If "None" is not expected, `tpm_policy = args.get("tpm_policy", {})` can work. If args is a config, then `fallback={}`.

Reply:
Thanks!